src/utils.ts

```diff
@@ -297,3 +297,203 @@ export class BinaryHeap<T> {
 		return this._items.length
 	}
 }
+
+export const enum EnumRunesCode {
+	HIGH_SURROGATE_START = 0xd800,
+	HIGH_SURROGATE_END = 0xdbff,
+
+	LOW_SURROGATE_START = 0xdc00,
+
+	REGIONAL_INDICATOR_START = 0x1f1e6,
+	REGIONAL_INDICATOR_END = 0x1f1ff,
+
+	FITZPATRICK_MODIFIER_START = 0x1f3fb,
+	FITZPATRICK_MODIFIER_END = 0x1f3ff,
+
+	VARIATION_MODIFIER_START = 0xfe00,
+	VARIATION_MODIFIER_END = 0xfe0f,
+
+	DIACRITICAL_MARKS_START = 0x20d0,
+	DIACRITICAL_MARKS_END = 0x20ff,
+
+	SUBDIVISION_INDICATOR_START = 0x1f3f4,
+	TAGS_START = 0xe0000,
+	TAGS_END = 0xe007f,
+
+	ZWJ = 0x200d,
+}
+
+export const GRAPHEMES = Object.freeze([
+	0x0308, // ( ◌̈ ) COMBINING DIAERESIS
+	0x0937, // ( ष ) DEVANAGARI LETTER SSA
+	0x093F, // ( ि ) DEVANAGARI VOWEL SIGN I
+	0x0BA8, // ( ந ) TAMIL LETTER NA
+	0x0BBF, // ( ி ) TAMIL VOWEL SIGN I
+	0x0BCD, // ( ◌்) TAMIL SIGN VIRAMA
+	0x0E31, // ( ◌ั ) THAI CHARACTER MAI HAN-AKAT
+	0x0E33, // ( ำ ) THAI CHARACTER SARA AM
+	0x0E40, // ( เ ) THAI CHARACTER SARA E
+	0x0E49, // ( เ ) THAI CHARACTER MAI THO
+	0x1100, // ( ᄀ ) HANGUL CHOSEONG KIYEOK
+	0x1161, // ( ᅡ ) HANGUL JUNGSEONG A
+	0x11A8, // ( ᆨ ) HANGUL JONGSEONG KIYEOK
+]);
```

Comment:
I'm not knowledge in this field. Is there a reference to all these constants in `EnumRunesCode` and `GRAPHEMES`?